pallets/cash/src/chains.rs

```diff
@@ -386,10 +391,14 @@ impl Chain for Ethereum {
         hash
     }
 
-    fn recover_address(data: &[u8], signature: Self::Signature) -> Result<Self::Address, Reason> {
+    fn recover_user_address(data: &[u8], signature: Self::Signature) -> Result<Self::Address, Reason> {
         Ok(compound_crypto::eth_recover(data, &signature, true)?)
     }
 
+    fn recover_address(data: &[u8], signature: Self::Signature) -> Result<Self::Address, Reason> {
```

Comment:
the same as above?

Reply:
The `bool` flag indicates whether or not the signature comes from a user. That is: was the signature `sign(data)` or `sign("\x19Ethereum Signed Message:" + data.length.toString() + data")`. This was the problem that caused us to keep double signing since we recovered the wrong version in our code that checked if we already signed.